uiowa/uiowa
# ITS Alerts Digest Emails

Resolves #7373

<img width="643" alt="Screenshot 2024-03-04 at 2 48 30 PM" src="https://github.com/uiowa/uiowa/assets/4663676/6c93045b-dbbf-45dd-bbe8-a958377bbb83">

# Summary

This PR established a alerts digest drush command that will be scheduled and triggered in PROD. The command reuses the code put together for https://github.com/uiowa/uiowa/pull/7397 so some of that has been adjusted to handle both types of emails.

# To Test

```
ddev blt ds --site=its.uiowa.edu && ddev drush @its.local its-digest && ddev launch -m
```

- You should see terminal output that the Alert Digest was sent.
- You should see an email with the from information set correctly. The to value should be using the site email address since the env is not PROD.
- Subject line should match the current website's email: `IT Service Alerts Daily Digest`
- You should then see an output of alerts grouped by type (Alerts (Outages and Degradations), Planned Maintenance, Service Announcements, Ongoing Maintenance) depending on if content for those types exist.
- The email should not be sent if there are no alerts that meet the above types' criteria.
- The alerts should contain the date, linked title, updates if they exist and the description.
- At the bottom of the email should be related links.
- Enable `dblog` and trigger the drush command again. You should see a message related to the command here: https://its.uiowa.ddev.site/admin/reports/dblog
- Make sure the individual alert email is still working by triggering one of the alerts to send an email on save. The output should match.
- Special notes about dates - "On the hour" time representations should have a space between the value and the meridian (am/pm). Also, the email should include the end date if not ongoing (e.g. planned maintenance, closed alerts)
